src/plugins/plugin.tooltip.js

```diff
@@ -128,7 +128,9 @@ function createTooltipItem(chart, item) {
 		formattedValue: value,
 		dataset,
 		dataIndex: index,
-		datasetIndex
+		datasetIndex,
+		x: element.x,
```

Comment:
Do you need to update the migration guide to remove lines that say these were removed?

But can we update the sample instead of adding these back? I think you could do something like `chart.getMeta(datasetIndex).data[dataIndex].x`.  I have a couple concerns with adding these here:
* I worry about the performance impact. We've seen that creating objects is really expensive. I'd assume creating larger objects might be more expensive than creating smaller objects, but haven't tested that
* There's a lot of properties which might make it a little harder to find what you want
* `x` and `y` are kind of generic names. For example, it's not immediately clear that it refers to the element location and not the mouse location. And if you know it refers to the element location it's still not clear where on the element - the lower left corner, the center, etc. Plus I imagine some elements might not define `x` and `y`

Reply:
I'm not really worried about the performance, you have to interact for this to happen and there the expensive part is finding the right element(s). We have that quite well covered.
I first thought of updating the sample, but it is not straight forward. You need to get the parsed values and use scales to get the pixels. I do think tat adding properties to a object at creation time us not that expensive.
Also I'd think those were added for a use case.

Reply:
Was the removal of these properties in the migration guide? If so, we should update the migration guide as well

Reply:
Yes, removed that line from the migration guide.

Reply:
Another option could be attaching the element directly, that could cover some other use cases.
`chart.getMeta(datasetIndex).data[dataIndex].x` should work too, not sure why I thought of needing to use scales for those.

Reply:
Yeah, I think I'd prefer `element` to `x`/`y`. But it doesn't seem to be used very frequently, so I think the `chart.getMeta(datasetIndex).data[dataIndex]` longhand would maybe be just as good here. I don't have strong feelings though, so whatever you and @etimberg want to do is fine with me

Reply:
I'll wait for @etimberg to comment

Reply:
I have no strong preference on `x` and `y` in the tooltip. I've never used them myself. Attaching the element directly seems like a good idea to me

Reply:
Attached element directly